sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/IEDAdapter.java

```diff
@@ -375,10 +380,42 @@ public Optional<TCompasICDHeader> getCompasICDHeader() {
 
     /**
      * Get value of private type COMPAS-SystemVersion
+     *
      * @return COMPAS-SystemVersion private value if present, else empty Optional
      */
     public Optional<TCompasSystemVersion> getCompasSystemVersion() {
         return PrivateService.extractCompasPrivate(currentElem, TCompasSystemVersion.class);
     }
 
+    /**
+     * Update and/or create Monitoring LNs (LSVS and LGOS) into LDSUIED for each bound ExtRef of each LDevice
+     *
+     * @return a list of SclReport Objects that contains errors
+     */
+    public List<SclReportItem> manageMonitoringLns() {
+        Optional<LDeviceAdapter> optionalLDeviceAdapter = findLDeviceAdapterByLdInst(LDSUIED);
+        if (optionalLDeviceAdapter.isPresent()) {
+            LDeviceAdapter lDeviceAdapter = optionalLDeviceAdapter.get();
+            List<SclReportItem> sclReportItems = new ArrayList<>();
+            sclReportItems.addAll(lDeviceAdapter.manageMonitoringLns(retrieveAllExtRefForServiceType(TServiceType.GOOSE), DO_GOCBREF, TSystemLNGroupEnum.LGOS));
+            sclReportItems.addAll(lDeviceAdapter.manageMonitoringLns(retrieveAllExtRefForServiceType(TServiceType.SMV), DO_SVCBREF, TSystemLNGroupEnum.LSVS));
+            return sclReportItems;
+        } else {
+            return Collections.emptyList();
+        }
```

Comment:
Use functional capabilities of `Oprtional`
```suggestion
                return findLDeviceAdapterByLdInst(LDSUIED).map(lDeviceAdapter -> {
                    List<SclReportItem> sclReportItems = new ArrayList<>();
                    sclReportItems.addAll(lDeviceAdapter.manageMonitoringLns(retrieveAllExtRefForServiceType(TServiceType.GOOSE), DO_GOCBREF, TSystemLNGroupEnum.LGOS));
                    sclReportItems.addAll(lDeviceAdapter.manageMonitoringLns(retrieveAllExtRefForServiceType(TServiceType.SMV), DO_SVCBREF, TSystemLNGroupEnum.LSVS));
                    return sclReportItems;
                })
                .orElseGet(Collections::emptyList);
```

or you go full functional style (but maybe less readable) : 
```suggestion
        return findLDeviceAdapterByLdInst(LDSUIED)
                .stream()
                .flatMap(lDeviceAdapter -> Stream.concat(
                        lDeviceAdapter.manageMonitoringLns(retrieveAllExtRefForServiceType(TServiceType.GOOSE), DO_GOCBREF, TSystemLNGroupEnum.LGOS).stream(),
                        lDeviceAdapter.manageMonitoringLns(retrieveAllExtRefForServiceType(TServiceType.SMV), DO_SVCBREF, TSystemLNGroupEnum.LSVS).stream()))
                .toList();
```

Reply:
As return value of findLDeviceAdapterByLdInst(LDSUIED) is an optional, i write the code this way (less code) : 
```suggestion
       List<SclReportItem> sclReportItems = new ArrayList<>();
        findLDeviceAdapterByLdInst(LDSUIED).ifPresent(lDeviceAdapter -> {
            sclReportItems.addAll(lDeviceAdapter.manageMonitoringLns(retrieveAllExtRefForServiceType(TServiceType.GOOSE), DO_GOCBREF, TSystemLNGroupEnum.LGOS));
            sclReportItems.addAll(lDeviceAdapter.manageMonitoringLns(retrieveAllExtRefForServiceType(TServiceType.SMV), DO_SVCBREF, TSystemLNGroupEnum.LSVS));
        });
        return sclReportItems;
```

Reply:
OK